src/main/java/thaumicboots/api/IBoots.java

```diff
@@ -50,29 +49,33 @@ default void setIsInertiaCanceling(ItemStack stack, boolean enabled) {
         stack.stackTagCompound.setBoolean(TAG_MODE_INERTIA, enabled);
     }
 
-    default double changeSpeed(ItemStack stack) {
+    default double changeSpeed(ItemStack stack, double modifier) {
         if (stack.stackTagCompound == null) {
             stack.setTagCompound(new NBTTagCompound());
         }
         // Internally MC returns 0 by default if the tag is not present, we do not need a presence check.
         double oldSpeed = stack.stackTagCompound.getDouble(TAG_MODE_SPEED);
-        double newSpeed = oldSpeed + Config.bootsSpeedChangeRate;
-        if (newSpeed > 1) {
+        double newSpeed = ((Math.round(20D * (oldSpeed + modifier)))) / 20D;
+        if (newSpeed > 1.05D) {
             newSpeed = 0;
+        } else if (newSpeed > 1) {
+            newSpeed = 1;
```

Comment:
What is this change doing?

Reply:
These two changes are to set the value of the modifier to 1 if it is less than .05(the minimum increment value) over 1, but to otherwise loop back around to 0. The intent is if you have the modifier at a value like .95 that it doesn't immediately loop back to 0 and skip 1. I do realize as I'm writing this that there is a better way to do this logic though, so I will change that accordingly.